### DISCUSSION TITLE
Geopandas not installing as a package when pip installing in editable (`-e`) mode.

### DISCUSSION THREAD

**Discussion by spolloni** [geopandas/geopandas#2736] [Installation]
- [x] I have read the [documentation on installation](https://geopandas.org/install.html) and followed the instructions provided.

- [x] I have looked through [issues labeled "installation"](https://github.com/geopandas/geopandas/labels/installation) in the geopandas repo.

Maybe this is the intended behavior, but in following the [contributing](https://geopandas.org/en/stable/community/contributing.html#making-a-development-build) instructions I am noticing `geopandas` doesn't pip install as an importable package in editable mode (`-e`), and am finding this to be a pain point.

```{bash}
git clone https://github.com/geopandas/geopandas.git
python -m venv .venv
source .venv/bin/activate
pip install pandas fiona shapely pyproj rtree pytest
pip install -e ./geopandas
```

Then:
```{
❯ python -c "import geopandas as gpd; print(dir(gpd))"
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

Whereas:
```
❯ cd geopandas &&  python -c "import geopandas as gpd; print(dir(gpd))"
['GeoDataFrame', 'GeoSeries', ..., 'tools']
```

A bit of context: while writing code I encountered a strange behavior which I suspect is tied to geopandas, a dependency of my project. It would have been nice to `pip install -e` geopandas and keep using it as a package while investigating this behavior in my specific setup.

**Comment by martinfleis** [comment]
Hi, not sure where this comes from. `pip install -e .` normally works for me on macOS and Ubuntu. I assume this may be some sort of a mixup within import preference as Python first imports a folder with a given them if it is in the working dir and then looks for installed packages.

**Comment by jorisvandenbossche** [comment]
Normally, python import should give precedence to actually (editable or not) installed packages over non-package directories in the current working directory. 

For example, if I run `python -c "import geopandas as gpd; print(dir(gpd))"` from the parent directory where the geopandas clone is located, it does import the editable install, not the directory. 
So I _think_ that means that something went wrong during the installation. 

Could you show the output of `pip install -e ./geopandas` ? And does `pip list` show geopandas?
